src/gsy_e/models/strategy/energy_parameters/energy_params_eb.py

```diff
@@ -1,16 +1,157 @@
 from logging import getLogger
-from typing import Dict
+from typing import Dict, Optional
 
 import pendulum
+from gsy_framework.utils import convert_kW_to_kWh
+
+from gsy_e.gsy_e_core.enums import AvailableMarketTypes
+from gsy_e.models.state import LoadState
+from gsy_e.models.strategy.energy_parameters.standard_profile_builders import StandardProfileParser
 
 logger = getLogger(__name__)
 
 
+class StandardProfileException(Exception):
+    """Exception for the StandardProfile."""
+
+
+class LoadSSPEnergyParameters:
+    """Energy parameters of the load strategy for the forward markets."""
+
+    _STANDARD_SOLAR_PROFILE = StandardProfileParser.parse()
+
+    def __init__(self, capacity_kW):
+        self.state = LoadState()
+        self.capacity_kW: float = capacity_kW
+
+        self._area = None
+        self._profile_generator: Optional[ForwardTradeProfileGenerator] = None
+
+    def serialize(self):
+        """Return dict with the current energy parameter values."""
+        return {
+            "capacity_kW": self.capacity_kW
+        }
+
+    def event_activate_energy(self, area):
+        """Update energy requirement upon the activation event."""
+        self._area = area
+        self._profile_generator = ForwardTradeProfileGenerator(
+            peak_kWh=convert_kW_to_kWh(self.capacity_kW, self._area.config.slot_length))
+
+    def event_traded_energy(self, energy_kWh, market_slot: pendulum.DateTime, product_type):
+        """
+        When a trade happens, we want to split the energy through the entire year following a
+        standard solar profile.
+        """
+        assert self._profile_generator is not None
+        # Create a new profile that spreads the trade energy across multiple slots. The values
+        # of this new profile are obtained by scaling the values of the standard solar profile
+        return self._profile_generator.generate_trade_profile(
+            energy_kWh=energy_kWh,
```

Comment:
Interesting, therefore in this case you are using the peak value in order to scale the whole profile, am I correct ? I will implement it first and then maybe ask some potential optimization here, because the first thing that comes to mind is that finding this peak value might be more performance-intensive than necessary. I will come back to it if I see that the performance here becomes a problem, thanks!

Reply:
The steps I'm following are these:
1. Find the scaling factor as:
    ```py
    # `new_peak` is the `capacity_kWh` of the asset
    scaling_factor = new_peak / peak_of_SSP
    ```
2. Scale the original SSP using the scaling factor. We obtain the **"capacity profile"**.
3. Compute a new scaling factor for the traded energy: 
    ```py
    # `new_peak` is the energy traded by the asset
    scaling_factor = new_peak / peak_of_SSP
    ```
4. Scale the original SSP using the "trade" scaling factor. In this way, we obtain the profile we need to subtract from the energy that remains to be traded.
5. Subtract this profile from the **capacity profile** in order to know the residual energy that can be bid in follow-up trades.

Right now I'm a bit confused about the last two steps (4 and 5). Shouldn't (4) already represent the energy we want to trade? Why do we want to get the residual dictionary (5)?